tests/test_modeling_common.py

```diff
@@ -158,6 +159,28 @@ def test_determinism(self):
             max_diff = np.amax(np.abs(out_1 - out_2))
             self.assertLessEqual(max_diff, 1e-5)
 
+    def test_forward_signature(self):
+        config, _ = self.model_tester.prepare_config_and_inputs_for_common()
+
+        for model_class in self.all_model_classes:
+            model = model_class(config)
+            signature = inspect.signature(model.forward)
+            # signature.parameters is an OrderedDict => so arg_names order is deterministic
+            arg_names = [*signature.parameters.keys()]
+
+            if model.config.is_encoder_decoder:
+                expected_arg_names = [
+                    "input_ids",
+                    "attention_mask",
+                    "decoder_input_ids",
+                    "decoder_attention_mask",
+                    "encoder_outputs",
+                ]
+                self.assertListEqual(arg_names[:5], expected_arg_names)
+            else:
+                expected_arg_names = ["input_ids"]
```

Comment:
@LysandreJik @sgugger - I wanted to enforce that the signature is `["input_ids", "attention_mask"]` for non Seq2Seq models, but there were too many failing tests meaning that too many models have a different API and I'm not sure whether it's worth such a big breaking change so I put `attention_mask` out of the list.

![Screenshot from 2020-09-30 20-50-15](https://user-images.githubusercontent.com/23423619/94728943-4b999700-0361-11eb-8963-47bd35ac9065.png)

Reply:
Do we need the ordering to enable something for those models?

Reply:
no we don't, so I don't really think it's worth it to force `attention_mask` to be at 2nd position

Reply:
Actually, some models might not be able to use the `attention_mask` in torchscript the way it is implemented now

Reply:
I think this should be enforced in most models, I like this test! For example with RoBERTa, it seems that all architectures follow the `input_ids` into `attention_mask`, except the `RobertaForMultipleChoice`. I see this more as a bug than something voluntary.

Some models like GPT-2 however should opt-out, as they use the past key values as a second value, which makes sense and is a defining trait of their API.

Reply:
I would advocate for a case-by-case study of these models to see which should have the ordering with `input_ids` > `attention_mask`, and which should stay the way they are.

Reply:
Agree! Will open an issue about it after merging this PR

Reply:
Noted the signature ordering problem and some other problems in Issues #7510, #7511, #7512